Sources/Paywalls/Components/PaywallPackageComponent.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PaywallPackageComponent.swift
+//
+//  Created by Josh Holtz on 9/27/24.
+
+import Foundation
+
+// swiftlint:disable missing_docs
+
+#if PAYWALL_COMPONENTS
+
+public extension PaywallComponent {
+
+    struct PackageComponent: PaywallComponentBase {
+
+        let type: ComponentType
+        public let packageID: String
+        public let components: [PaywallComponent]
+
+        public init(type: ComponentType,
+                    packageID: String,
+                    isButton: Bool = true,
+                    components: [PaywallComponent]
+        ) {
+            self.type = .package
+            self.packageID = packageID
+            self.components = components
```

Comment:
Just for my understanding: these child components are things like text, which can contain various details of the package such as price? 

Should they be somehow limited to certain types, so people don't add something like a PurchaseButton inside a Package? Or do we want to allow for maximum flexibility?

Reply:
Yeah I agree... Maybe we can define protocols for these different types and only accept those types of protocols? My main concern with that is we won't have compile safety inside when trying to distinguish between the different subtypes of the protocol...

Reply:
> Should they be somehow limited to certain types, so people don't add something like a PurchaseButton inside a Package? Or do we want to allow for maximum flexibility?

This type of validation will happen in both the frontend and the backend! Eventually 😇 There is no validation there yet but as long as the frontend and backend have the schema with these rules, the SDK should be good 💪